app/src/services/tag.js

```diff
@@ -98,24 +99,42 @@ const service = {
    * @function dissociateTags
    * dissociates all provided tags from a versionId
    * @param {string} versionId The uuid id column from version table
-   * @param {string[]} [keys=undefined] array of tag keys (eg ['animal', 'colour'])
+   * @param {object[]} [tags=undefined] array of tags (eg: [{ key: 'a', value: '1'}, {key: 'B', value: ''}])
    * @param {object} [etrx=undefined] An optional Objection Transaction object
    * @returns {Promise<number>} The result of running the delete operation (number of rows deleted)
    * @throws The error encountered upon db transaction failure
    */
-  dissociateTags: async (versionId, keys = undefined, etrx = undefined) => {
+  dissociateTags: async (versionId, tags = undefined, etrx = undefined) => {
     let trx;
     try {
       trx = etrx ? etrx : await Tag.startTransaction();
-      let response = [];
-
-      // delete joining records
-      response = await VersionTag.query(trx)
-        .allowGraph('tag')
-        .withGraphJoined('tag')
-        .whereIn('tag.key', keys)
-        .modify('filterVersionId', versionId)
-        .delete();
+      let response = 0;
+      let count = 0;
```

Comment:
Count should be instantiated within the async tag foreach callback function instead of being scoped one level up. It isn't being used anywhere else outside of that.